AmplifyPlugins/Predictions/AWSPredictionsPlugin/Service/Predictions/AWSPredictionsService+Transcribe.swift

```diff
@@ -31,7 +31,17 @@ extension AWSPredictionsService: AWSTranscribeStreamingServiceBehavior {
         request.mediaSampleRateHertz = 8_000
 
         transcribeClientDelegate.connectionStatusCallback = { status, error in
-            if status == .connected {
+            if status == .closed && error != nil {
+                guard error != nil else {
+                    return
+                }
+                let nsError = error as NSError?
+                guard nsError?.domain != NSURLErrorDomain else {
+                    let predictionsError = PredictionsErrorHelper.mapError(nsError!)
```

Comment:
sorry for the iterations on this, but this still doesn't make sense to me because it's an 
1. explicit check on URLError
2. If it is any type of URLError, always generate a PredictionsError and send that back to the caller. from traversing the logic, it will send Predictions.network when URLError is related to connectivity, otherwise send Predictions.unknown

Now there are two places to generate a PredicationsError, in `mapError(_ error: NSError)` and `mapPredictionsServiceError(_ error: NSError)`, how do we as plugin developers know which one to call use?

From the tests, I see that the intent is to only see an PredictionsError on the failed event when the URLError is `notConnectedToInternet` and `cannotFindHost` which means you are missing a test for "Failed event when error is URLError but is not one of the following: notConnectedToInternet or cannotFindHost"

If you want to return an unknown error anytime it's an URLError that is not related to connectivity then you should add a test in there as well.

But i _think_ you are only care to notify the caller on network related URLErrors, then the code should be

```swift
if let urlError = error as? URLError?, (case .notConnected = urlError.code || case .cannotfindHost  == urlErorr.code) {
     // generate PredictionsError generically
    let predictionsError = PredictionsErrorHelper.mapServiceError(error)
   onEvent(.failed(predictionsError)
  }

}
```

Reply:
```swift
let predictionsError = PredictionsErorrHelper.mapServiceError(error)
if case .network = predictionsError {
   onEvent(.failed(predictionsError)
}
}
```